Extensions/Emails/Services/EmailService.cs

```diff
@@ -0,0 +1,36 @@
+using System.Collections.Generic;
+using System.Linq;
+using Lombiq.HelpfulExtensions.Extensions.Emails.Models;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using OrchardCore.Email;
+using OrchardCore.Environment.Shell.Scope;
+
+namespace Lombiq.HelpfulExtensions.Extensions.Emails.Services
+{
+    public class EmailService : IEmailService
+    {
+        public void SendEmailDeferred(EmailParameters parameters) =>
+            ShellScope.AddDeferredTask(async scope =>
+            {
+                var smtpService = scope.ServiceProvider.GetService<ISmtpService>();
+                var logger = scope.ServiceProvider.GetService<ILogger<EmailService>>();

```

Comment:
It's enough to request this service when you need it, which should be rare.
`ILogger<T>` is injected via factory so it's not zero cost.